### PR TITLE
Added database controller to BankController

### PR DESCRIPTION
**Pull request by Koen03**

### CODE REVIEW

File: Bank/src/main/java/com/ark/bank/BankController.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
     private final Set<BankTransaction> bankTransactions = new HashSet<>();
     private final ICentralBankConnection centralBankConnection;
     private int defaultSessionTime = 900000;
-    public DatabaseController db;
+    private DatabaseController db;
```
Comment (ArthurDoorgeest) [inline Bank/src/main/java/com/ark/bank/BankController.java]: rename to databaseController

File: Bank/src/main/java/com/ark/bank/BankController.java [hunk 2]
```diff
@@ -34,12 +34,11 @@ public BankController(String bankId, ICentralBankConnection centralBankConnectio
         }
 
         this.bankId = bankId;
-        connectDB();
         this.centralBankConnection = centralBankConnection;
     }
 
-    private void connectDB(){
-        this.db = new DatabaseController(this.bankId);
+    public void connectDB(String id){
```
Comment (ArthurDoorgeest) [inline Bank/src/main/java/com/ark/bank/BankController.java]: Rename to connectDatabase and String bankId

File: Bank/src/main/java/com/ark/bank/BankController.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
     private final Set<BankTransaction> bankTransactions = new HashSet<>();
     private final ICentralBankConnection centralBankConnection;
     private int defaultSessionTime = 900000;
+    private DatabaseController BankDatabaseConnection;
```
Comment (rickvanderheijden) [inline Bank/src/main/java/com/ark/bank/BankController.java]: Rename to databaseController... use the class name as much as possible.

File: Bank/src/main/java/com/ark/bank/BankController.java [hunk 4]
```diff
@@ -36,6 +37,10 @@ public BankController(String bankId, ICentralBankConnection centralBankConnectio
         this.centralBankConnection = centralBankConnection;
     }
 
+    public void connectToBankDatabase(String id){
+        this.BankDatabaseConnection = new DatabaseController(id);
```
Comment (rickvanderheijden) [inline Bank/src/main/java/com/ark/bank/BankController.java]: Makr onnectToDatabase a boolean method. Check is string not null or empty. rename id to bankId (we use that everywhere)

File: Bank/src/main/java/com/ark/bank/BankHost.java [hunk 5]
```diff
@@ -28,6 +28,6 @@ public static void main(String[] args) throws RemoteException {
         IBankController bankController = new BankController(BankId, centralBankConnection);
         CentralBankConnector centralBankConnector = new CentralBankConnector(bankController, BankId, URLBase);
         GUIConnector guiConnector = new GUIConnector(bankController);
-        Persistence database = new Persistence(BankId);
+        ((BankController) bankController).connectToBankDatabase(BankId);
```
Comment (rickvanderheijden) [inline Bank/src/main/java/com/ark/bank/BankHost.java]: Why do y ou need to cast?? Did you not change the interface?

File: Bank/src/main/java/com/ark/bank/DatabaseController.java [hunk 6]
```diff
@@ -11,18 +11,18 @@
...
 
     private EntityManager entityManager;
 
     /**
-     * Creates a instance of Persistence object
+     * Creates a instance of DatabaseController object
      * @param BankId The bank id of the current bank.
      */
-    public Persistence(String BankId){
+    public DatabaseController(String BankId) {
```
Comment (rickvanderheijden) [inline Bank/src/main/java/com/ark/bank/DatabaseController.java]: BankId should be bankId

File: Bank/src/test/java/unittest/TestDatabaseController.java [hunk 7]
```diff
@@ -15,14 +15,14 @@
...
+public class TestDatabaseController {
 
-    Persistence p;
+    DatabaseController p;
     Customer c1;
 
     @Before
     public void setUp() {
-        p = new Persistence("Test");
+        p = new DatabaseController("TEST");
```
Comment (rickvanderheijden) [inline Bank/src/test/java/unittest/TestDatabaseController.java]: p ?  --> databaseController

File: Bank/src/test/java/unittest/TestDatabaseController.java [hunk 8]
```diff
@@ -15,14 +15,14 @@
 /**
  * @author Koen Sengers
  */
-public class TestPersistence {
+public class TestDatabaseController {
 
-    Persistence p;
+    DatabaseController p;
     Customer c1;
```
Comment (rickvanderheijden) [inline Bank/src/test/java/unittest/TestDatabaseController.java]: c1 --> customerOne or customer1

File: Bank/src/test/java/unittest/TestDatabaseController.java [hunk 9]
```diff
@@ -241,4 +241,78 @@ public void testGetPersistTransactionNULL(){
         List<BankTransaction> tr = p.getPersistTransaction("RABO123456789");
         assertEquals(0, tr.size());
     }
+
+    @Test
+    public void testGetAllCustomers(){
+        Customer c2 = new Customer("Rick", "Beek en Donk", "Hates Mac");
```
Comment (rickvanderheijden) [inline Bank/src/test/java/unittest/TestDatabaseController.java]: c2 --> customerTwo of customer2
c3 --> customerThree of customer2

File: Bank/src/test/java/unittest/TestDatabaseController.java [hunk 10]
```diff
@@ -241,4 +241,78 @@ public void testGetPersistTransactionNULL(){
...
+        Customer c3 = new Customer("Arthur", "Zaltbommel", "Also hates Mac");
+        p.persist(c1);
+        p.persist(c2);
+        p.persist(c3);
+
+        List<Customer> l = p.getAllCustomers();
```
Comment (rickvanderheijden) [inline Bank/src/test/java/unittest/TestDatabaseController.java]: l --> this is the result of your test right? Name it result.

**Pull request merged**